$OpenBSD: patch-src_up-main_c,v 1.1.1.1 2011/05/23 19:04:57 landry Exp $
glib-unix.h and g_unix_signal_add_watch_full appeared in glib 2.29.5
--- src/up-main.c.orig	Sun May 22 22:35:09 2011
+++ src/up-main.c	Sun May 22 22:35:14 2011
@@ -41,7 +41,7 @@
 #include "up-kbd-backlight.h"
 #include "up-wakeups.h"
 
-#if GLIB_CHECK_VERSION(2,28,7)
+#if GLIB_CHECK_VERSION(2,29,5)
  #include <glib-unix.h>
 #endif
 
@@ -92,7 +92,7 @@ out:
 	return ret;
 }
 
-#if GLIB_CHECK_VERSION(2,28,7)
+#if GLIB_CHECK_VERSION(2,29,5)
 
 /**
  * up_main_sigint_cb:
@@ -201,7 +201,7 @@ main (gint argc, gchar **argv)
 		goto out;
 	}
 
-#if GLIB_CHECK_VERSION(2,28,7)
+#if GLIB_CHECK_VERSION(2,29,5)
 	/* do stuff on ctrl-c */
 	g_unix_signal_add_watch_full (SIGINT,
 				      G_PRIORITY_DEFAULT,
